### PR TITLE
Add implementation for ProfileForm and ProfileSummary

### PR DESCRIPTION
**Pull request by lucasecdb**
#### What problem is this solving?
This PR adds the initial implementation for the `ProfileForm` component, and adds a `ProfileSummary` component to show the preview of the profile inside `orderForm`.

[Related clubhouse story](https://app.clubhouse.io/vtex/story/32825/criar-componentes-de-profile-form-e-profile-preview)

#### How should this be manually tested?
[Workspace](https://chkio--checkoutio.myvtex.com/checkout#/profile)

#### Checklist/Reminders

- [x] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [x] Linked this PR to a Clubhouse story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Screenshots or example usage
image

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements 

#### Notes

If you have any suggestions on how I can improve the state management of the form, **please do so**, because I didn't really like handling it manually like I did here :-/

### CODE REVIEW

**Review by darlenedms — DISMISSED** [review]

**Comment by jeffersontuc** [comment]
@lucasecdb I'm receiving `400` at submitting profile data. Is this expected?

image

**Comment by lucasecdb** [comment]
@jeffersontuc it's not. can you share the error message?

**Comment by jeffersontuc** [comment]
@lucasecdb I edited my comment with the image showing the error.

File: docs/README.md [hunk 1]
```diff
@@ -1 +1,44 @@
...
+
+```json
+{
+  "dependencies": {
+    "vtex.checkout-profile": "0.x"
+  }
+}
+```
+
+> If your are using TypeScript, you may also want to run `vtex setup` after the step above
```
Comment (thiagomurakami) [inline docs/README.md]:
```suggestion
> If you are using TypeScript, you may also want to run `vtex setup` after the step above
```

File: react/ProfileForm.tsx [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+  }
+
+  const handleOptinNewsletterChange: React.ChangeEventHandler<HTMLInputElement> = evt => {
+    setOptInNewsletter(evt.target.checked)
+  }
+
+  const handleChange: React.ChangeEventHandler<HTMLInputElement> = evt => {
+    dispatch({
+      type: 'update',
+      field: evt.target.name as keyof ProfileState,
```
Comment (thiagomurakami) [inline react/ProfileForm.tsx:212]: Can you use `as const` here?

File: react/ProfileForm.tsx [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+        ...profile,
+        [fieldName]: profileData[fieldName as keyof ProfileState].value,
+      }),
+      {} as any
+    )
+
+    const [
+      { success: profileUpdateSuccess },
+      { success: clientPreferencesUpdateSuccess },
+    ] = await Promise.all([
```
Comment (thiagomurakami) [inline react/ProfileForm.tsx]: Don't you need a `try...catch` block here?

**Comment by lucasecdb** [comment]
@jeffersontuc can you look again?

**Review by thiagomurakami — CHANGES_REQUESTED** [review]

**Comment by thiagomurakami** [comment]
> If you have any suggestions on how I can improve the state management of the form, please do so, because I didn't really like handling it manually like I did here :-/

We could use a form library like `formik`, but I don't know if it's overkill or not haha

File: docs/README.md (hunk 1 above)
Comment (lucasecdb) [reply docs/README.md]: nice catch!

File: react/ProfileForm.tsx (hunk 3 above)
Comment (lucasecdb) [reply react/ProfileForm.tsx]: hmm, I think so. I'll add it

File: react/ProfileForm.tsx (hunk 2 above)
Comment (lucasecdb) [reply react/ProfileForm.tsx:212]: sure!

**Comment by lucasecdb** [comment]
> We could use a form library like `formik`, but I don't know if it's overkill or not haha

yeah, I thought about using it! I just wanted a better way to handle the form so I can kill that giant reducer hahah

we may need to look into it in the future, because even though we only have 4 fields in this form, it will grow to much more when we include the optional company fields (like CNPJ and friends)

File: react/ProfileForm.tsx (hunk 2 above)
Comment (lucasecdb) [reply react/ProfileForm.tsx:212]: TS didn't like this change :-/

```
[tsserver 2322] [E] Type 'string' is not assignable to type '"firstName" | "lastName" | "phone" | "documentType" | "document"'.

[tsserver 1355] [E] A 'const' assertions can only be applied to references to enum members, or string, number, boolean, array, or object literals.
```

**Review by thiagomurakami — APPROVED** [review]

**Review by jeffersontuc — APPROVED** [review]

**Pull request merged**